source/simulator/melt.cc

```diff
@@ -230,81 +288,70 @@ namespace aspect
-                                                         :
-                                                         eta * 2.0/3.0 * (scratch.div_phi_u[i] * scratch.div_phi_u[j])
-                                                        )
-                                                      - (pressure_scaling *
-                                                         scratch.div_phi_u[i] * scratch.phi_p[j])
-                                                      // finally the term -div(u). note the negative sign to make this
-                                                      // operator adjoint to the grad(p) term
-                                                      - (pressure_scaling *
-                                                         scratch.phi_p[i] * scratch.div_phi_u[j])
-                                                      +
-                                                      (- pressure_scaling * pressure_scaling / viscosity_c
-                                                       * scratch.phi_p_c[i] * scratch.phi_p_c[j])
-                                                      - pressure_scaling * scratch.div_phi_u[i] * scratch.phi_p_c[j]
-                                                      - pressure_scaling * scratch.phi_p_c[i] * scratch.div_phi_u[j]
-                                                      - K_D * pressure_scaling * pressure_scaling *
-                                                      (scratch.grad_phi_p[i] * scratch.grad_phi_p[j])
-                                                      + (this->get_material_model().is_compressible()
-                                                         ?
-                                                         K_D * pressure_scaling * pressure_scaling / density_f
-                                                         * scratch.phi_p[i] * (scratch.grad_phi_p[j] * density_gradient_f)
-                                                         :
-                                                         0.0))
-                                                    * scratch.finite_element_values.JxW(q);
-                      }
-                }
+              if (rebuild_stokes_matrix)
+                for (unsigned int j=0; j<stokes_dofs_per_cell; ++j)
+                  {
+                    data.local_matrix(i,j) += ( (use_tensor ?
+                                                 eta * 2.0 * (scratch.grads_phi_u[i] * stress_strain_director * scratch.grads_phi_u[j])
+                                                 :
+                                                 eta * 2.0 * (scratch.grads_phi_u[i] * scratch.grads_phi_u[j]))
+                                                - (use_tensor ?
+                                                   eta * 2.0/3.0 * (scratch.div_phi_u[i] * trace(stress_strain_director * scratch.grads_phi_u[j]))
```

Comment:
I guess it'd be ok if you said you don't care too much and leave it to those people who care about melt :)